tests/test_operations.py

```diff
@@ -0,0 +1,119 @@
+import random
+
+random.seed(0)
+import numpy as np
+
+np.random.seed(0)
+import torch
+from torch.autograd import grad
+
+torch.manual_seed(0)
+torch.use_deterministic_algorithms(True)
+from pyqtorch.ansatz import AlternateLayerAnsatz
+from pyqtorch.core.batched_operation import batchedRY
+from pyqtorch.core.circuit import QuantumCircuit
+from pyqtorch.core.measurement import total_magnetization
+from pyqtorch.core.operation import CNOT, RY
+from torch.nn import Module, ModuleList
+
+
+class TestFM(QuantumCircuit):
+    def __init__(self, n_qubits: int = 4):
+        super().__init__(n_qubits)
+        self.qubits = range(n_qubits)
+
+    def forward(self, state: torch.Tensor, x: torch.Tensor) -> torch.Tensor:
+        for i in self.qubits:
+            state = RY(x, state, [i], self.n_qubits)
+        return state
+
+
+class TestBatchedFM(QuantumCircuit):
+    def __init__(self, n_qubits: int = 4):
+        super().__init__(n_qubits)
+        self.qubits = range(n_qubits)
+
+    def forward(self, state: torch.Tensor, x: torch.Tensor) -> torch.Tensor:
+        for i in self.qubits:
+            state = batchedRY(x[:, 0], state, [i], self.n_qubits)
+        return state
+
+
+class TestNetwork(Module):
+    def __init__(self, network, n_qubits=4) -> None:
+        super().__init__()
+        self.n_qubits = n_qubits
+        self.network = ModuleList(network)
+        self.state = QuantumCircuit(n_qubits)
+        self.operator = total_magnetization
+
+    def forward(self, nx: torch.Tensor) -> torch.Tensor:
+        batch_size = len(nx)
+        state = self.state.init_state(batch_size=batch_size, device=nx.device)
+        for layer in self.network:
+            state = layer(state, nx)
+        return self.operator(state, self.n_qubits, batch_size)
+
+
+# TODO: these are all the same test, would be better to parameterize a test
+def test_batched_network():
+    ansatz = AlternateLayerAnsatz(n_qubits=4, n_layers=4)
+    network = TestNetwork([TestFM(), ansatz])
+    batched_network = TestNetwork([TestBatchedFM(), ansatz])
+    # to ensure the parameters are the same
+    batch_size = 2
```

Comment:
@awennersteen I do not see how this actually ensures that the parameters are the same. The quantum circuits are still initialized with different parameters as far as I understand.

Reply:
I only create the `ansatz` onece, i.e. it has one set of params.
We don't update these in this example. This is what ensures that they are the same. I will add that

The feature maps don't have any nn.Parameters, only `x` dependencies. Hence no parameters are initialized.

Reply:
@awennersteen I see, thanks.